helm/adelphi/templates/cassandra-mgmt-dockerfile-configmap.yaml

```diff
@@ -5,222 +5,81 @@ metadata:
-                cluster_name \
-                endpoint_snitch \
-                listen_address \
-                num_tokens \
-                rpc_address \
-                start_rpc \
-            ; do
-                var="CASSANDRA_${yaml^^}"
-                val="${!var}"
-                if [ "$val" ]; then
-                    _sed-in-place "$CASSANDRA_CONF/$CASSANDRA_YAML" \
-                        -r 's/^(# )?('"$yaml"':).*/\2 '"$val"'/'
-                fi
-            done
-
-            for rackdc in dc rack; do
-                var="CASSANDRA_${rackdc^^}"
-                val="${!var}"
-                if [ "$val" ]; then
-                    _sed-in-place "$CASSANDRA_CONF/cassandra-rackdc.properties" \
-                        -r 's/^('"$rackdc"'=).*/\1 '"$val"'/'
-                fi
-            done
-      fi
-
-      MGMT_API_ARGS=""
-
-      # Hardcoding these for now
-      MGMT_API_CASSANDRA_SOCKET="--cassandra-socket /tmp/cassandra.sock"
-      MGMT_API_LISTEN_TCP="--host tcp://0.0.0.0:8080"
-      MGMT_API_LISTEN_SOCKET="--host file:///tmp/oss-mgmt.sock"
-
-      MGMT_API_ARGS="$MGMT_API_ARGS $MGMT_API_CASSANDRA_SOCKET $MGMT_API_LISTEN_TCP $MGMT_API_LISTEN_SOCKET"
-
-      # These will generally come from the k8s operator
-      if [ ! -z "$MGMT_API_EXPLICIT_START" ]; then
-        MGMT_API_EXPLICIT_START="--explicit-start $MGMT_API_EXPLICIT_START"
-        MGMT_API_ARGS="$MGMT_API_ARGS $MGMT_API_EXPLICIT_START"
-      fi
-
-      if [ ! -z "$MGMT_API_TLS_CA_CERT_FILE" ]; then
-        MGMT_API_TLS_CA_CERT_FILE="--tlscacert $MGMT_API_TLS_CA_CERT_FILE"
-        MGMT_API_ARGS="$MGMT_API_ARGS $MGMT_API_TLS_CA_CERT_FILE"
-      fi
-      if [ ! -z "$MGMT_API_TLS_CERT_FILE" ]; then
-        MGMT_API_TLS_CERT_FILE="--tlscert $MGMT_API_TLS_CERT_FILE"
-        MGMT_API_ARGS="$MGMT_API_ARGS $MGMT_API_TLS_CERT_FILE"
-      fi
-      if [ ! -z "$MGMT_API_TLS_KEY_FILE" ]; then
-        MGMT_API_TLS_KEY_FILE="--tlskey $MGMT_API_TLS_KEY_FILE"
-        MGMT_API_ARGS="$MGMT_API_ARGS $MGMT_API_TLS_KEY_FILE"
-      fi
-
-      if [ ! -z "$MGMT_API_PID_FILE" ]; then
-        MGMT_API_PID_FILE="--pidfile $MGMT_API_PID_FILE"
-        MGMT_API_ARGS="$MGMT_API_ARGS $MGMT_API_PID_FILE"
-      fi
-
-      MGMT_API_CASSANDRA_HOME="--cassandra-home /var/lib/cassandra/"
-      MGMT_API_ARGS="$MGMT_API_ARGS $MGMT_API_CASSANDRA_HOME"
-
-      if [ ! -z "$MGMT_API_NO_KEEP_ALIVE" ]; then
-        MGMT_API_NO_KEEP_ALIVE="--no-keep-alive $MGMT_API_NO_KEEP_ALIVE"
-        MGMT_API_ARGS="$MGMT_API_ARGS $MGMT_API_NO_KEEP_ALIVE"
-      fi
-
-      MGMT_API_JAR="$(find "/opt/mgmtapi" -name *server*.jar)"
-
-      echo "Running" java ${MGMT_API_JAVA_OPTS} -Xms128m -Xmx128m -jar "$MGMT_API_JAR" $MGMT_API_ARGS
-      exec /tini -g -- gosu cassandra java ${MGMT_API_JAVA_OPTS} -Xms128m -Xmx128m -jar "$MGMT_API_JAR" $MGMT_API_ARGS
-
-    fi
-
-    exec "$@"
\ No newline at end of file
+{{ .Files.Get "config/mgmt-api/docker-entrypoint.sh" | indent 4 }}
```

Comment:
Should we consider doing something like this with the Dockerfile as well?  I love the idea of getting this out of inline YAML.